### ISSUE TITLE
onboarding

### ISSUE THREAD

**Issue by snankani** [snankani/AzurePipelinesTest#25]
Goal of the onboarding checklist is to help new hubber or new member in the team to have access to necessary information which will enable them to learn in a self-reliant way and set their foundation in the team to accelerate in their career.
Information will be shared in the form of predefined steps to be followed, links pointing to documentation, videos, or presentation.

**WEEK-1** 

Welcome to GitHub!!!

Most of your time, in your first week, will be spent in attending global virtual onboarding sessions. So we will keep your first week light in terms of team specific ramp up. 

To make your onboarding easier, please reach out to your manager and buddy to get started at GitHub.

- [ ]  Talk to your Manager (recommend topics - how to get started, buddy details, ramp up plan) Hopefully this is already done, if not then please reach out to your manager
- [ ]  Talk to your Buddy. Hopefully this is already done, if not then please reach out to your buddy

**KNOW THE PRODUCT**

Great way to start learning is to gain an understanding of GitHub as a product. In week-1, the goal will be to get a high-level understanding of the product by doing a hands-on exercise of most frequently used features. 
As part of global onboarding sessions, you should have already created an account in GitHub and tried most frequently used features like creation of PR and GitHub issues and others.

If you are part of a team which actively works on Azure DevOps then take some time to practice below steps.
- [ ] [Create an organization on Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/create-organization?view=azure-devops)
- [ ] [Create a project](https://docs.microsoft.com/en-us/azure/devops/organizations/projects/create-project?view=azure-devops&tabs=preview-page) and a sample Git and TFVC repo
- [ ] [Create PR](https://docs.microsoft.com/en-us/azure/devops/repos/git/pullrequest?view=azure-devops) and add someone from your team to review the PR, merge the PR once approved
- [ ] [Create couple of work items](https://docs.microsoft.com/en-us/azure/devops/boards/backlogs/add-work-items?view=azure-devops&tabs=browser)

**TEAM PORTFOLIO**

This should cover the list of products owned by the team. List of products where the team is making active investments.

**TECH STACK** :

 
KNOW THE PEOPLE
You will be working with the team. Knowing your team composition and the team members will be a nice way to start your first week at GitHub.
https://team.githubapp.com/org is the place to find Organizational hierarchy
[ ] Please enter your name or GitHub alias in the search box to find the team structure/composition along with their GitHub alias
[ ] Select your team member one at a time to find their name, designation and location

[ ] Send your intro along with photo to your manager (re-validate to see if this is required since new norm of welcoming the hubber is on slack channel)

KNOW THE PROCESS
<Team specific>
How does the team work? 
Please add details of team Scrum, retrospective meeting, sprint cycle, knowledge sharing session and other relevant information. DLs, Slack and Teams channel which are used by the team.
India level channels to be added
Think of this to cover recurring meetings that a team usually has.
</Team specific>
[ ] Look for any mandatory training and take time to complete those (<TODO> update this with a link)
[ ] Work with your buddy to get added to Slack and Team channels used by team on a daily or frequent basis
[ ] Work with your buddy to get added to Team DLs used by team on a daily or frequent basis
[ ] Reach out to buddy/manager to raise a request to include you in all the team level recurring meetings
[ ] Start participating in scrum meetings from week-1, if you can else participating from week-2 is also good

 
WEEK-2
Welcome to week 2 in GitHub !!
Tip: Reach out to your buddy whenever you get stuck or have a query and they will help you unblock or redirect to an appropriate person or point to a document to unblock you.
KNOW THE PRODUCT
Hope you have a decent understanding of GitHub products or ADO and high-level knowledge of the products/areas owned by the team.

<Team specific> 
As part of week-1, hope you had a chance to set up your organization and project in Azure DevOps and explored features like PR and work item creation. 
This week, you will focus on learning about Pipelines vertical and GitHub Actions. 
Azure DevOps 
Architecture  (Video duration: 2 HRS)
Pipelines 
Deployment Groups
Deployment Pools
Environments
ARM Service Connection and SPN flow
Pipeline Tasks
Pipeline Tasks Extension
GitHub Actions

Create one pipeline and one workflow using any of the task/action from this list
Start learning Git, Typescript, c#, powershell

Vision/roadmap document
Architectural/Design document/video
Steps to setup Dev Environment (GitHub / ADO)
Debug the code at a high level 
[ ] Tech specific training <team to add recommended documentation or videos or courses> links to recorded brownbag session 
All of this can spill over to week-3 and that is completely fine
</Team specific>
 
KNOW THE PEOPLE
By this time you should have learned about your team and team composition. This week, you can make an effort to get to know each of your team members better. 
[ ] Talk to your buddy/manager to learn more about your stakeholders - PM/Designer/Other team or team members 
[ ] Reach out to your team member on Slack and set up a one-on-one meeting with your peer to exchange a brief intro and know them better. 
Repeat the above step with possibly every member of your team by the end of third week from your start date on GitHub. 
[ ]  Talk to your Manager (recommend topic(s) - ramp up progress)
[ ]  Consistently keep talking to your Buddy and it is completely fine and acceptable to reach out to Buddy for very small things to get clarified. Please do not hesitate. 
KNOW THE PROCESS
[ ] Look for any mandatory training and take time to complete those (<TODO> update this with a link)
 
WEEK-3
Welcome to week 3 in GitHub !!
Tip: Reach out to your buddy whenever you get stuck or have a query and they will help you unblock or redirect to an appropriate person or point to a document to unblock you.
KNOW THE PRODUCT
<Team specific> Deployment process, prod setup in terms of scale unit, regions and other details
DRI related things like videos, or presentation or wiki links, DRI duties and responsibilities
Get yourself added to DRI cycle, shadow with other engineer in 4th week or later and then be prepared to be DRI anytime from your 6th week 
Know your team OKR
 
KNOW THE PEOPLE
<TODO>  Other interesting channels this is covered in github global onboarding
[ ]  Talk to your Manager (recommend topics - role and responsibilities)
KNOW THE PROCESS
<Team specific> How the work is assigned, tracked ?? what is the working mode - like spec review, development cycle, bug bash, deployment process (this can go to third week), basically product development cycle. 
<TODO> Mention ongoing GitHub India initiatives and provide link to GitHub repo and documentation which can help the engineer to nominate themselves in those initiatives based on their interest
[ ] Capture the feedback in this document/GitHub issue and share the same with buddy/manager on what can be improved to make the onboarding process more effective (<TODO> update the document link or mention the GitHub issue)
